probe_src/probe_frontend/python/ops.py

```diff
@@ -0,0 +1,361 @@
+
+@dataclass(init=False)
+class StatOp:
+    path: Path
+    flags: int
+    statx_buf: statx
+    ferrno: int
+    def __init__(self, path: Path, flags: int, statx_buf: statx, ferrno: int):
+        self.path = Path(**path)
+        self.flags = int(flags)
+        self.statx_buf = statx(**statx_buf)
+        self.ferrno = int(ferrno)
+
+@dataclass(init=False)
+class ReaddirOp:
+    dir: Path
+    child: bytes
+    all_children: bool
+    ferrno: int
+    def __init__(self, dir: Path, child: bytes, all_children: bool, ferrno: int):
+        self.dir = Path(**dir)
+        self.child = bytes(child)
+        self.all_children = bool(all_children)
+        self.ferrno = int(ferrno)
+
+@dataclass(init=False)
+class WaitOp:
+    pid: int
+    options: int
+    status: int
+    ret: int
+    ferrno: int
+    def __init__(self, pid: int, options: int, status: int, ret: int, ferrno: int):
+        self.pid = int(pid)
+        self.options = int(options)
+        self.status = int(status)
+        self.ret = int(ret)
+        self.ferrno = int(ferrno)
+
+@dataclass(init=False)
+class GetRUsageOp:
+    waitpid_arg: int
+    getrusage_arg: int
+    usage: rusage
+    ferrno: int
+    def __init__(self, waitpid_arg: int, getrusage_arg: int, usage: rusage, ferrno: int):
+        self.waitpid_arg = int(waitpid_arg)
+        self.getrusage_arg = int(getrusage_arg)
+        self.usage = rusage(**usage)
+        self.ferrno = int(ferrno)
+
+@dataclass(init=False)
+class ReadLinkOp:
+    path: Path
+    resolved: bytes
+    ferrno: int
+    def __init__(self, path: Path, resolved: bytes, ferrno: int):
+        self.path = Path(**path)
+        self.resolved = bytes(resolved)
+        self.ferrno = int(ferrno)
+
+@dataclass(init=False)
+class Metadata:
+    @dataclass(init=False)
+    class Mode:
+        mode: int
+        def __init__(self, mode: int):
+            self.mode = int(mode)
+
+    @dataclass(init=False)
+    class Ownership:
+        uid: int
+        gid: int
+        def __init__(self, uid: int, gid: int):
+            self.uid = int(uid)
+            self.gid = int(gid)
+
+    @dataclass(init=False)
+    class Times:
+        is_null: bool
+        atime: timeval
+        mtime: timeval
+        def __init__(self, is_null: bool, atime: timeval, mtime: timeval):
+            self.is_null = bool(is_null)
+            self.atime = timeval(**atime)
+            self.mtime = timeval(**mtime)
+
+    value: typing.Union[Mode, Ownership, Times]
+    def __init__(self, **kwargs: typing.Mapping[str, typing.Any]):
+        if len(kwargs) != 1:
+            raise ValueError("Malformed Enum constructor args")
+        key = list(kwargs.keys())[0]
+        if key in self.__class__.__dict__:
+            self.value = self.__class__.__dict__[key](**kwargs[key])
+        else:
+            self.value = mod.__dict__[key](**kwargs[key])
+
+@dataclass(init=False)
+class UpdateMetadataOp:
+    path: Path
+    flags: int
+    metadata: Metadata
```

Comment:
See comment on JSON decoding for why we should be able to get away with this.
```suggestion
    metadata: Mode | Ownership | Times
```

Reply:
See my response to that comment for an explanation of the technical limitations of the current process and my plan(s) to address them.

Reply:
solved with type aliasing in https://github.com/charmoniumQ/PROBE/pull/20/commits/0beca52947f8d3c46cf527b04411f50ad76e39fe.